packages/api/src/controllers/managementPlanController.js

```diff
@@ -206,11 +206,97 @@ const managementPlanController = {
   delManagementPlan() {
     return async (req, res) => {
       try {
-        const isDeleted = await ManagementPlanModel.query()
+        const { management_plan_id } = req.params;
+
+        const managementPlan = await ManagementPlanModel.query()
           .context(req.auth)
-          .where({ management_plan_id: req.params.management_plan_id })
-          .delete();
-        if (isDeleted) {
+          .where({ management_plan_id })
+          .where('deleted', false)
+          .first();
+
+        if (!managementPlan) {
+          return res.status(404).send('Management plan not found');
+        }
+
+        const result = await ManagementPlanModel.transaction(async (trx) => {
```

Comment:
I think this transaction object should be passed around: https://vincit.github.io/objection.js/guide/transactions.html#passing-around-a-transaction-object

Reply:
I took this verbatim from the existing abandonment controller so I assumed it was correct 🙂 I'll take a close read through that documentation -- looks at first glance like both patterns are permissible.

Reply:
Okay, read it and added 😄 Thank you!

Do you think the abandonment should be updated now too? Or a separate ticket?

Reply:
definitely separate ticket lol

Reply:
I would like to understand the status of this comments. In the same controller we have 2 different ways of handling a transaction commit/rollback. Ideally we use one pattern. I'm not sure if just passing `trx` ensures commit at the end?

Reply:
Yes, passing `trx` is enough for commit or rollback (it's pattern #2 in the documentation @Duncan-Brain linked above and the pattern in the primary example). We have plenty of both patterns in the codebase. I did a quick search and found an [older Jira ticket](https://lite-farm.atlassian.net/browse/LF-355) suggesting we should move towards the passing `trx` pattern, but we have a complete mix, so not sure if getting them aligned is at all in the scope of this ticket.